docs/versioning.md

```diff
@@ -0,0 +1,42 @@
+# Kubernetes API and Release Versioning
+
+Legend:
+
+* **Kube &lt;major&gt;.&lt;minor&gt;.&lt;patch&gt;** refers to the version of Kubernetes that is released. This versions all components: apiserver, kubelet, kubectl, etc.
+* **API vX[betaX]** refers to the version of the HTTP API.
+
+## Release Timeline
+
+### Minor version timeline
+
+* Kube 1.0
+* Kube 1.0.x: We create a 1.0-patch branch and backport critical bugs and security issues to it. Patch releases occur as needed.
+* Kube 1.1-alpha1: Cut from HEAD, smoke tested and released two weeks after Kube 1.0's release. Every two weeks a new alpha is released from HEAD.
+* Kube 1.1-beta1: When HEAD is feature complete, we create a 1.1-snapshot branch and release it as a beta. (The 1.1-snapshot branch may be created earlier if something that definitely won't be in 1.1 needs to be merged to HEAD.) This should occur 6-8 weeks after Kube 1.0. Development continues at HEAD and only fixes are backported to 1.1-snapshot.
+* Kube 1.1-rc1: Released from 1.1-snapshot when it is considered stable and ready for testing. Most users should be able to upgrade to this version in production.
+* Kube 1.1: Final release. Should occur between 3 and 4 months after 1.0.
+
+### Major version timeline
+
+There is no mandated timeline for major versions. They only occur when we need to start the clock on deprecating features. A given major version should be the latest major version for at least one year from its original release date.
+
+## Upgrade policies
+
+* Users can upgrade from any Kube 1.x release to any other Kube 1.x release as a rolling upgrade across their cluster.
```

Comment:
#4855: Actual rolling upgrade is a non-goal for 1.0. We won't stop it, but we're not writing it.